### PR TITLE
Ferveo Variant docs - also removal of existing stale ferveo docs.

### LINKED ISSUE #140: Document simple and precomputed variants and their benefits/drawbacks in ferveo library documentation
See comment here about content: https://github.com/nucypher/nucypher/pull/3174#issuecomment-1631165980

Related to https://github.com/nucypher/nucypher/pull/3174

### PR DESCRIPTION
**Pull request by derekpierre**
**Type of PR:**
- [ ] Bugfix
- [ ] Feature
- [X] Documentation
- [ ] Refactor
- [ ] Other

**Required reviews:** 
> How many reviews does the PR author need?
- [ ] 1
- [X] 2
- [ ] 3

**What this does:**
- Adds documentation for Ferveo variants and their characteristics
- Re-use pre-existing ferveo docs set up, but also remove stale data that is no longer applicable based on our own changes.

**Issues fixed/closed:**
Fixes #140 .

**Why it's needed:**
> Explain how this PR fits in the greater context of the NuCypher Network.
> E.g., if this PR address a `nucypher/productdev` issue, let reviewers know!

**Notes for reviewers:**
> What should reviewers focus on? 
> Is there a particular commit/function/section of your PR that requires more attention from reviewers?

I tried to keep some of the more relevant existing Ferveo docs. I have no problems with just deleting them if they are just no longer applicable. 

Let me know what you think.

### CODE REVIEW

**Review by piotr-roslaniec — APPROVED** [review]
LGTM :tada: 

I think it's sufficiently descriptive at this time. Thank you for putting this to paper.

File: book/src/variants.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+This is the case where you can concurrently contact `n` nodes and simply wait until any `m` nodes respond with decryption shares.
+
+### Advantages
+- Any (arbitrary) `m` decryption shares out of `n` can be requested, obtained, and combined
+- Any singular unresponsive node does not prevent the ability to combine returned *simple* decryption shares
+- Only one request round is needed if done concurrently i.e. send `n` request, and wait for `m` replies. A total of
+  `n - m + 1` nodes would have to not respond for the failure case.
+
+### Disadvantages
+- Since the `m` nodes are arbitrary, combining the decryption shares requires the requester to do a more computationally intensive operation.
```
[the remark this answers is not in the document]
Comment (derekpierre) [reply book/src/variants.md:26]: If there is something to link to - yep, no problem, just point me to it.

**Comment by derekpierre** [comment]
Side note: there is a mention of `simple` and `fast` here - https://github.com/derekpierre/nucypher-ferveo/blob/variant-docs/book/src/tpke.md#threshold-decryption-fast-method - are these the same concepts or something separate?

File: book/src/variants.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (piotr-roslaniec) [reply book/src/variants.md:26]: In the PRs description: https://github.com/nucypher/ferveo/pull/32

**Comment by piotr-roslaniec** [comment]
@derekpierre Yes, we use the original `simple` variant. `fast` is also implemented (still present in the codebase), but remains unused.

File: book/src/variants.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (derekpierre) [reply book/src/variants.md:26]: Those look like scripts to do the benchmarking, is there an output file  containing the benchmarks somewhere or no? If not, perhaps you can add instructions on running those scripts to the docs in a separate PR.

[the remark this answers is not in the document]
Comment (piotr-roslaniec) [reply book/src/variants.md:26]: This is the output of `cargo bench` and these files are generated locally. We could add a section dedicated to benchmarking with these instructions and other information.

**Review by cygnusv — APPROVED** [review]

File: book/src/variants.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (derekpierre) [reply book/src/variants.md:26]: Let's do that in a different PR.

**Pull request merged**